libparsec/crates/platform_http_proxy/src/lib.rs

```diff
@@ -56,15 +56,15 @@ mod native {
 
     impl ProxyConfig {
         /// Set the http proxy to use, will overwrite the previous configuration.
-        pub(crate) fn with_http_proxy(mut self, proxy: String) -> anyhow::Result<Self> {
+        pub fn with_http_proxy(mut self, proxy: String) -> anyhow::Result<Self> {
```

Comment:
If we expose those method, we must implement them in wasm, which is currently not possible given in wasm the proxy config is always a noop...

From what I understand, you use the proxy object in the SSE tests to tell the client to go to the test proxy server.
However this test proxy server is already a TCP proxy, so it should be fine to just modify `alice_device`'s organization addr to use the TCP proxy domain/port instead of the testbed one.

Reply:
I tried to mutate `alice` with the following snippet:

```rust
    let mut alice = env.local_device("alice@dev1");
    // ...

    let alice_proxy = proxy::spawn(env.server_addr.clone()).await.unwrap();

    std::sync::Arc::get_mut(&mut alice)
        .expect("We don't have the exclusive copy of `alice`")
        .organization_addr = BackendOrganizationAddr::new(
        alice_proxy.backend_addr(),
        bob.organization_id().clone(),
        bob.root_verify_key().clone(),
    );
```

Sadly this snippet panic at the ``.expect("We don't have the exclusive copy of `alice`")``'s line.

Help wanted

Reply:
you don't need to modify the device returned by `env.local_device("alice@dev1");` given the device object is not used internally by the testbed

Basically you could be creating your own `LocalDevice` object (think for example of a test where we try to load the device from a file and then start the client from this).

Hence you can use `Arc::make_mut(&mut alice)` to create a new alice device with the patched addr, then create the authenticated cmds with this device

Reply:
OK

Reply:
done

Reply:
Why do you need in public ?

Reply:
That modification was reverted, strange that github don't indicate it.